tests/integration/standard/test_query.py

```diff
@@ -1175,3 +1179,211 @@ def test_unicode(self):
+
+        ddl = '''
+                CREATE TABLE {0}.{1} (
+                    k int PRIMARY KEY,
+                    v int )'''.format(cls.ks_name, cls.table_name)
+        cls.session.execute(ddl)
+        ddl = '''
+                CREATE TABLE {0}.{1} (
+                    k int PRIMARY KEY,
+                    v int )'''.format(cls.alternative_ks, cls.table_name)
+        cls.session.execute(ddl)
+
+        cls.session.execute("INSERT INTO {}.{} (k, v) VALUES (1, 1)".format(cls.ks_name, cls.table_name))
+        cls.session.execute("INSERT INTO {}.{} (k, v) VALUES (2, 2)".format(cls.alternative_ks, cls.table_name))
+
+    @classmethod
+    def tearDownClass(cls):
+        ddl = "DROP KEYSPACE {}".format(cls.alternative_ks)
+        cls.session.execute(ddl)
+        ddl = "DROP KEYSPACE {}".format(cls.ks_name)
+        cls.session.execute(ddl)
+        cls.cluster.shutdown()
+
+class QueryKeyspaceTests(BaseKeyspaceTests):
+
+    def test_setting_keyspace(self):
+        self._check_set_keyspace_in_statement(self.session)
+
+    def test_setting_keyspace_and_session(self):
+        cluster = Cluster(protocol_version=ProtocolVersion.V5, allow_beta_protocol_version=True)
+        session = cluster.connect(self.alternative_ks)
+        self.addCleanup(cluster.shutdown)
+
+        self._check_set_keyspace_in_statement(session)
+
+    def test_setting_keyspace_and_session_after_created(self):
+        cluster = Cluster(protocol_version=ProtocolVersion.V5, allow_beta_protocol_version=True)
+        session = cluster.connect()
+        self.addCleanup(cluster.shutdown)
+
+        session.set_keyspace(self.alternative_ks)
+        self._check_set_keyspace_in_statement(session)
+
+    def test_setting_keyspace_and_same_session(self):
+        cluster = Cluster(protocol_version=ProtocolVersion.V5, allow_beta_protocol_version=True)
+        session = cluster.connect(self.ks_name)
+        self.addCleanup(cluster.shutdown)
+
+        self._check_set_keyspace_in_statement(session)
+
+
+@greaterthanorequalcass40
+class SimpleWithKeyspaceTests(QueryKeyspaceTests, unittest.TestCase):
+    def test_lower_protocol(self):
+        cluster = Cluster(protocol_version=ProtocolVersion.V4)
+        session = cluster.connect(self.ks_name)
+        self.addCleanup(cluster.shutdown)
+
+        simple_stmt = SimpleStatement("SELECT * from {}".format(self.table_name), keyspace=self.ks_name)
+        # This raises cassandra.cluster.NoHostAvailable: ('Unable to complete the operation against
+        # any hosts', {<Host: 127.0.0.3 datacenter1>: UnsupportedOperation('Keyspaces may only be
+        # set on queries with protocol version 5 or higher. Consider setting Cluster.protocol_version to 5.',),
+        # <Host: 127.0.0.2 datacenter1>: ConnectionException('Host has been marked down or removed',),
+        # <Host: 127.0.0.1 datacenter1>: ConnectionException('Host has been marked down or removed',)})
+        with self.assertRaises(NoHostAvailable):
```

Comment:
Should setting the keyspace in `SimpleStatement` raise `UnsupportedOperation` or `UnsupportedOperation` for old versions? As mentioned in a previous comment, raising something would break the API, since I believe now the keyspace is just ignored-